blob/s3blob/s3blob.go

```diff
@@ -395,6 +398,32 @@ func (b *bucket) NewRangeReader(ctx context.Context, key string, offset, length
 	}, nil
 }
 
+// etagToMD5 processes an ETag header and returns an MD5 hash if possible.
+// S3's ETag header is sometimes a quoted hexstring of the MD5. Other times,
+// notably when the object was uploaded in multiple parts, it is not.
+// We do the best we can.
+// Some links about ETag:
+// https://docs.aws.amazon.com/AmazonS3/latest/API/RESTCommonResponseHeaders.html
+// https://github.com/aws/aws-sdk-net/issues/815
+// https://teppen.io/2018/06/23/aws_s3_etags/
+func eTagToMD5(etag *string) []byte {
+	if etag == nil {
+		// No header at all.
+		return nil
+	}
+	// Strip the expected leading and trailing quotes.
+	quoted := *etag
+	if quoted[0] != '"' || quoted[len(quoted)-1] != '"' {
+		return nil
+	}
+	unquoted := quoted[1 : len(quoted)-1]
+	// Un-hex; we return nil on error. In particular, we'll get an error here
+	// for multi-part uploaded blobs, whose ETag will look something like
+	// "aaaaaaa-2".
```

Comment:
So there's no way a multi-part ETag can ever look like hex? If it weren't for the `-`, your example would be. 

I'm hoping your comment can say (truthfully) "multi-part upload ETags always contain `-`, so they can never be legal hex" or something like that.

Reply:
I believe that is true, so I made the comment stronger.